torch/csrc/jit/codegen/cuda/README.md

```diff
@@ -0,0 +1,215 @@
+# NVFuser - A Fusion Code Generator for NVIDIA GPUs
+_NVFuser is integrated as a backend for TorchScript's Profiling Graph Executor_
+
+## Enabling NVFuser
+_NVFuser is not currently the default fuser for NVIDIA GPUs._
+
+**Fusions will only show up during the ~3rd iteration of execution, the exact number depends on profiling executor's optimization phases**
+
+### Enable by Context Manager
+
+```
+jit_model = torch.jit.script(model)
+
+with torch.jit.fuser("fuser2") :
+    for _ in range(5) :
+        outputs = jit_model(inputs)
+```
+
+### Enable by Specific Functions
+
+1. Disable cpu/gpu fusion for native/nnc fuser
+```
+torch._C._jit_override_can_fuse_on_cpu(False)
+torch._C._jit_override_can_fuse_on_gpu(False)
+```
+2. Disable nnc fuser
+```
+torch._C._jit_set_texpr_fuser_enabled(False)
+```
+3. Enable nvfuser
+```
+torch._C._jit_set_nvfuser_enabled(True)
+```
+
+## Simple knobs to change fusion behavior
+
+1. Allow single node fusion `torch._C._jit_set_nvfuser_single_node_mode(True)`
+Fusion group is only created when two or more compatible ops are grouped together. Turn on single node fusion would allow fusion pass to create fusion group with a single node, this is very handy for testing and could be useful when single node generated kernel out-performs native cuda kernels in framework.
+
+2. Allow horizontal fusion `torch._C._jit_set_nvfuser_horizontal_mode(True)`
+Fusion pass fuses producer to consumer, horizontal mode allows sibling nodes that shared tensor input to be fused together. This could save input memory bandwidth.
+
+3. Turn off guard for fusion `torch._C._jit_set_nvfuser_guard_mode(False)`
+This disables the runtime check on fusion group pre-assumptions (tensor meta information / constant inputs / profiled constants), this really is only used for testing as we want to ensure generated kernels are indeed tested and you should avoid using this in training scripts.
+
+## Fusion Debugging
```

Comment:
Thanks for this doc. The debugging procedures documented here would be really helpful. 

It seems that you're suggesting a step-by-step approach, which starts with inspecting the user-given function in TorchScript. Perhaps, it would be even more helpful to make the steps more explicit and emphasize what the user should pay attention to at each step.

I'm not sure how much effort we would need to spend here, but FAQs would be a great addition.

Reply:
Let me take another look at the debugging steps. Those docs are mostly just recording how I extract smaller repros from user provided scripts, hoping that it could be helpful to external users as well. I'll try to be more explicit on instructions.

> I'm not sure how much effort we would need to spend here, but FAQs would be a great addition.

An FAQs are a good section to have! We can have it there and fill it up as we get more feedback/questions.